### PR TITLE
Task Simple-singolo 

### PR DESCRIPTION
**Pull request by Thesilkroaduser**
<h1> Presentation of the finished website "Simple-singolo" </h1>

<h2> Main features: </h2>

1. Visual similarity of the site and layout
2. The site is adapted for 3 types of devices:
   - 768px <= screen width < 1020px 
   - 375px <= screen width < 768px 
   - screen width < 375px
3. Interactivity

<h2> Desktop version </h2>

<h3> Header: </h3>

   - Interactive navigation bar
   - The logo contains 2 different text elements, white and red
   - There is just one "h1" element on the page

> +10 points

<h3> Slider : </h3>

   - Interactive arrows

> +10 points

<h3> Services: </h3>

   - Flexible positioning

> +10 points

<h3> Portfolio: </h3>

   - Interactive navigation buttons
   - Four-column layout with images
   - If you add more elements with images to the grid (for example, 15), then those following the 12th aren't shown (13, 14, 15 are not visible).

> +10 points

<h3> Footer: </h3>

   - The copyright symbol (©) is text
   - Interactive icons

> +10 points

<h2>Responsive layout using media queries</h2>

<h3>768px <= screen  width < 1020px</h3>

<h4>Header</h4>

1. The logo on the left and the last navigation menu item on the right maintain their padding to the edges of the header block.

> +10 points

<h4>Slider </h4>

1. Phone images are scaled down to the screen width 

> +5 points

<h4>Our services</h4>

1. A three-column grid becomes a two-column grid
2. Distances change both from grid elements to block edges and the distance between adjacent elements

> +10 points

<h4>Portfolio</h4>

1. A four-column grid becomes a three-column grid

> +10 points

<h4>Footer</h4>

1. Copyright on the left and extreme social nets  icon on the right maintain their padding to the edges of the footer block

> +5 points

<h3>375px <= screen  width < 768px</h3>

<h4>Header </h4>

1. The logo remains in the center. The top navigation menu is hidden in the burger-menu on the left

> +10 points

<h4>Slider </h4>

1. Phone images are scaled down to the screen width 

> +5 points

<h4>Our services</h4>

1. A two-column grid becomes one column

> +10 points

<h4>Portfolio</h4>

1. A three-column grid becomes a two-column grid, 4 rows high, the rest of the elements are hidden
2. The copyright disappears from the page. Social icons networks are centered in the footer block

> +10 points

<h4>Footer</h4>

1. The copyright disappears from the page. Social icons networks are centered in the footer block

> +5 points

<h3>Screen width < 375px </h3>

<h4>Margins and block sizes are respected up to a threshold width of 320 pixels</h4>

> +10 points

<h2> Interactivity </h2>

<h3>Header</h3>

1. The selected menu item remains active, the previous one becomes inactive. At the same time, the page should move along the anchors to the specified position with smooth animation. When switching and scrolling, the menu element of the block on which the current position is located should become active.
2. When the screen width is less than 768px, the top navigation menu is hidden in the burger-menu on the left. By clicking on the burger-menu, a panel on the left is displayed and overlaps the application screen. The menu block takes up the entire height. The sandwich icon is rotated like on the layout.

> +20 points

<h3>Slider</h3>

1. When you press the slider control buttons, the slides switch with smooth animation.
2. This action can be done endlessly.

> +20 points

<h3>Portfolio</h3>

1. When switching tabs at the top of the block, the new tab becomes active.
2. Pictures change their position in random order

> +10 points

> The only feature that I did not implement is the distribution of the color of the blocks to the full width of the screen, with a device width of more than 1020 pixels, I sincerely apologize for that, I forgot.

> **TOTAL SCORE: 190 points**

**_Thanks for your attention! I am open to criticism, I am looking forward for your reply)_**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,144 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
```
Comment (gegorov) [inline index.html]: path should be relative 
```suggestion
    <link rel="stylesheet" href="./style.css">
```

File: index.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    <title>Document</title>
+</head>
+<body>
+    <div class="header-wrapper">
```
Comment (gegorov) [inline index.html]: why do you need this wrapper? all styles can be applied to class .header. Moreover, you are duplicating some styles in those two classes

File: index.html [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        <header class="header">
+            <nav class="burger-wrapper">
+                <a href="##" class="burger-opening"><img src="img/Burger_menu_icon/menu.svg" alt="menu icon" class="burger-icon"></a>
+                <nav class="burger-menu__hidden">
```
Comment (gegorov) [inline index.html]: why are you nesting navs? and why there are three navs on the page?

File: index.html [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                    <a href="##" class="burger-opening vertical__hidden"><img src="img/Burger_menu_icon/menu.svg" alt="menu icon" class="burger-icon"></a>
```
Comment (gegorov) [inline index.html]: why do you use ## inside of href? isn't one # enough for the anchor to work?

File: index.html [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <title>Document</title>
+</head>
+<body>
+    <div class="header-wrapper">
+        <header class="header">
+            <nav class="burger-wrapper">
+                <a href="##" class="burger-opening"><img src="img/Burger_menu_icon/menu.svg" alt="menu icon" class="burger-icon"></a>
```
Comment (gegorov) [inline index.html]: path should be relative and file&folder names should be lowercase

File: index.html [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                        <li class="burger__list-item"><a href="#slider" class="burger__link" id="burger-home">home</a></li>
+                        <li class="burger__list-item"><a href="#services" class="burger__link" id="burger-serv">services</a></li>
+                        <li class="burger__list-item"><a href="#portfolio" class="burger__link" id="burger-port">portfolio</a></li>
+                        <li class="burger__list-item"><a href="#" class="burger__link" id="burger-about">about</a></li>
+                        <li class="burger__list-item"><a href="#footer" class="burger__link" id="burger-contact">contacts</a></li>
+                    </ul>
+                </nav>
+            </nav>
+            <h1 class="logo">Singolo</h1>
+            <nav class="menu">
```
Comment (gegorov) [inline index.html]: keep it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). you do not need two identical menu. everything can be done with css

File: index.html [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                    <li class="list-item"><a href="#footer" class="list-item__link" id="contact">Contacts</a></li>
+                </ul>
+            </nav>
+        </header>
+    </div>
+    <main>
+        <div class="slider main-section" id="slider">
+            <div class="slider__wrapper">
+              <div class="slider__item">
+                <div class="item-content"><img src="img/Slides/Slide-1.png" alt="Slide with phone picture" class="content__image"></div>
```
Comment (gegorov) [inline index.html]: i think that one of the divs is redundant. do not create unnecessary nesting and div hell

File: index.html [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+              </div>
+              <div class="slider__item">
+                <div class="item-content"><img src="img/Slides/Slide-2.png" alt="Slide with phone picture" class="content__image"></div>
```
Comment (gegorov) [inline index.html]: again, files and folders should be lowercase

Comment (gegorov) [reply index.html]: and relative path

File: index.html (hunk 7 above)
Comment (gegorov) [reply index.html]: path and filenames

File: index.html [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+              </div>
+            </div>
+            <a class="slider__control slider__control_left" href="#" role="button"></a>
+            <a class="slider__control slider__control_right" href="#" role="button"></a>
+        </div>
+        <section class="services main-section" id="services">
+            <div class="services__header">
```
Comment (gegorov) [inline index.html]: redundant div

File: index.html [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+                <h2 class="section-title">Our Services</h2>
+                <p class="section-description">Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum. Duis mollis, non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh.</p>
+            </div>
+            <div class="services__content">
+                <div class="content-item">
+                    <img src="img/Services/services-item-1.svg" class="item__icon"></img>
```
Comment (gegorov) [inline index.html]: folder name

File: index.html [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+                    <div>    
+                        <h3 class="item__title">Custom Design</h3>
+                        <p class="item__description">Curabitur vestibulum eget mauris quis laoreet. Phasellus in quam laoreet, viverra lacus ut, ultrices velit.</p>
+                    </div>            
+                </div>
+                <div class="content-item">
+                    <img src="img/Services/services-item-2.svg" class="item__icon"></img>
```
Comment (gegorov) [inline index.html]: folder name

File: index.html [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+                    <img src="img/Services/services-item-6.svg" class="item__icon"></img>
+                    <div>    
+                        <h3 class="item__title">Feel Like A Star</h3>
+                        <p class="item__description">Quisque hendrerit purus dapibus, ornare nibh vitae, viverra nibh. Fusce vitae aliquam tellus.<p>
+                    </div>
+                </div>
+            </div>
+        </section>
+        <section class="portfolio main-section" id="portfolio">
+            <div class="porfolio__header">
```
Comment (gegorov) [inline index.html]: no need for this div

File: index.html [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+                <h2 class="section-title">Portfolio</h2>
+                <p class="section-description">Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit.</p>
+                <div class="categories">
+                    <a href="##" class="categories__item active_item" id="all">All</a>
+                    <a href="##" class="categories__item" id="web_design" role="button">Web Design</a>
+                    <a href="##" class="categories__item" id="graphic_design" role="button">Graphic Design</a>
+                    <a href="##" class="categories__item" id="artwork" role="button">Artwork</a>
```
Comment (gegorov) [inline index.html]: you can keep in this task, but in general, if you need a button you use <button> not <a> with role. it is not good from an accessibility point of view. and it is easier to handle actions with buttons

File: index.html [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+        </section>
+    </main>
+    <hr class="dividing-line">
+    <footer class="footer main-section" id="footer">
+        <p class="copyright">© Copyright 2020</p>
+        <div class="contacts">
+            <a href="##" class="contact__link"><img src="img/Contacts/ico-facebook.svg" alt="facebook icon" class="contacts__icon"></a>
+            <a href="##" class="contact__link"><img src="img/Contacts/ico-google-plus.svg" alt="google plus icon" class="contacts__icon"></a>
+            <a href="##" class="contact__link"><img src="img/Contacts/ico-twitter.svg" alt="twitter icon" class="contacts__icon"></a>
+            <a href="##" class="contact__link"><img src="img/Contacts/ico-linkedin.svg" alt="linkedin icon" class="contacts__icon"></a>
```
Comment (gegorov) [inline index.html]: usually such icons are not inserted as images but as glyphs from  [fonawesome](https://fontawesome.com/icons?d=gallery)

File: index.html [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+        </div>
+    </footer>
+    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js"></script>
+    <script src="slider.js"></script>
+    <script src="shuffle.js"></script>
+    <script src="scroll.js"></script>
+    <script src="burger.js"></script>
```
Comment (gegorov) [inline index.html]: relative path

File: shuffle.js [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+        container.appendChild(container.children[Math.random() * i | 0]);
+    }
+}
+
+all.addEventListener('click', () => {
+    web_design.classList.remove('active_item');
+    graphic_design.classList.remove('active_item');
+    artwork.classList.remove('active_item');
+    all.classList.add('active_item');
+    shuffle(shuffle_container);
```
Comment (gegorov) [inline shuffle.js:18]: you are repeating almost identical steps in all callbacks, so probably it is possible to make a function that handles all this class toggling stuff. and i guess it is enough to have one event listener on body, where you can play with event.target.

Comment (gegorov) [reply shuffle.js:18]: it is fine for the sake of the current task, but one you will study dom manipulation with js such solutions will not be acceptable

**Review by gegorov — CHANGES_REQUESTED** [review]
first, check my comments, and try to fix them on the correct file in simple-singolo folder. 
than remove duplicates
